src/lib/wallet.js

```diff
@@ -38,9 +38,30 @@ export default async function initSdk() {
 
   store.commit('setNodeStatus', NODE_STATUS_CONNECTING);
   try {
+    const { getMiddleware, initMiddleware } = useMiddleware({ store });
+    store.watch(
+      (state, getters) => getters.activeNetwork,
+      async (network, oldNetwork) => {
+        if (isEqual(network, oldNetwork)) return;
+        try {
+          if (store.state.nodeStatus === NODE_STATUS_ERROR && !store.getters['sdkPlugin/sdk']) {
```

Comment:
@peronczyk Could `nodeStatus` be one of the composables?
I see `store.commit('setNodeStatus', NODE_STATUS_CONNECTING);` too